test/src/test/java/org/ethereum/beacon/test/TestUtils.java

```diff
@@ -267,19 +281,43 @@ static BeaconChainSpec loadSpecByName(String name, boolean isBlsVerified) {
       Path dir,
       Class<? extends V> testsType,
       Function<Pair<TestCase, BeaconChainSpec>, Optional<String>> testCaseRunner,
-      Collection<String> exclusions) {
+      Collection<String> exclusions,
+      int threadNum) {
     List<File> files = getResourceFiles(dir.toString());
-    boolean failed = false;
+    AtomicBoolean failed = new AtomicBoolean(false);
+    System.out.println("Running tests in " + dir + " with " + threadNum + " thread(s)");
+    ThreadPoolExecutor executor = (ThreadPoolExecutor) Executors.newFixedThreadPool(threadNum);
+    AtomicInteger counter = new AtomicInteger(0);
+    CountDownLatch awaitLatch = new CountDownLatch(files.size());
     for (File file : files) {
-      System.out.println("Running tests in " + file.getName());
-      Optional<String> result = runAllTestsInFile(file, testCaseRunner, testsType, exclusions);
-      if (result.isPresent()) {
-        System.out.println(result.get());
-        System.out.println("\n----===----\n");
-        failed = true;
-      }
+      Runnable task =
+          () -> {
+            int num = counter.getAndIncrement();
+            System.out.println(num + ". Running tests in " + file.getName());
+            Optional<String> result =
+                runAllTestsInFile(file, testCaseRunner, testsType, exclusions);
+            if (result.isPresent()) {
+              System.out.println(num + ". " + result.get());
+              System.out.println(num + ". \n----===----\n");
+              failed.set(true);
+            }
+            awaitLatch.countDown();
+          };
+      executor.execute(task);
     }
-    assertFalse(failed);
+
+    try {
+      awaitLatch.await();
```

Comment:
You may use a default implementation of `Schedulers` interface to get an executor that returns `CompletableFuture`. It would be nice to combine them in a meta future to avoid this semaphore.